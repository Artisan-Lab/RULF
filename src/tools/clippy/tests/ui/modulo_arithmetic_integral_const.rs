#![warn(clippy::modulo_arithmetic)]
#![allow(
    clippy::no_effect,
    clippy::unnecessary_operation,
    clippy::modulo_one,
    clippy::identity_op
)]

fn main() {
    // Lint when both sides are const and of the opposite sign
    -1 % 2;
    1 % -2;
    (1 - 2) % (1 + 2);
    (1 + 2) % (1 - 2);
    35 * (7 - 4 * 2) % (-500 * -600);

    -1i8 % 2i8;
    1i8 % -2i8;
    -1i16 % 2i16;
    1i16 % -2i16;
    -1i32 % 2i32;
    1i32 % -2i32;
    -1i64 % 2i64;
    1i64 % -2i64;
    -1i128 % 2i128;
    1i128 % -2i128;
    -1isize % 2isize;
    1isize % -2isize;

    // No lint when both sides are const and of the same sign
    1 % 2;
    -1 % -2;
    (1 + 2) % (-1 + 2);
    (-1 - 2) % (1 - 2);

    1u8 % 2u8;
    1u16 % 2u16;
    1u32 % 2u32;
    1u64 % 2u64;
    1u128 % 2u128;
    1usize % 2usize;
}
