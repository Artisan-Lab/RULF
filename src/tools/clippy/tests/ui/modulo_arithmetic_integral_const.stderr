error: you are using modulo operator on constants with different signs: `-1 % 2`
  --> $DIR/modulo_arithmetic_integral_const.rs:11:5
   |
LL |     -1 % 2;
   |     ^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function
   = note: `-D clippy::modulo-arithmetic` implied by `-D warnings`

error: you are using modulo operator on constants with different signs: `1 % -2`
  --> $DIR/modulo_arithmetic_integral_const.rs:12:5
   |
LL |     1 % -2;
   |     ^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-1 % 3`
  --> $DIR/modulo_arithmetic_integral_const.rs:13:5
   |
LL |     (1 - 2) % (1 + 2);
   |     ^^^^^^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `3 % -1`
  --> $DIR/modulo_arithmetic_integral_const.rs:14:5
   |
LL |     (1 + 2) % (1 - 2);
   |     ^^^^^^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-35 % 300000`
  --> $DIR/modulo_arithmetic_integral_const.rs:15:5
   |
LL |     35 * (7 - 4 * 2) % (-500 * -600);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-1 % 2`
  --> $DIR/modulo_arithmetic_integral_const.rs:17:5
   |
LL |     -1i8 % 2i8;
   |     ^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `1 % -2`
  --> $DIR/modulo_arithmetic_integral_const.rs:18:5
   |
LL |     1i8 % -2i8;
   |     ^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-1 % 2`
  --> $DIR/modulo_arithmetic_integral_const.rs:19:5
   |
LL |     -1i16 % 2i16;
   |     ^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `1 % -2`
  --> $DIR/modulo_arithmetic_integral_const.rs:20:5
   |
LL |     1i16 % -2i16;
   |     ^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-1 % 2`
  --> $DIR/modulo_arithmetic_integral_const.rs:21:5
   |
LL |     -1i32 % 2i32;
   |     ^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `1 % -2`
  --> $DIR/modulo_arithmetic_integral_const.rs:22:5
   |
LL |     1i32 % -2i32;
   |     ^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-1 % 2`
  --> $DIR/modulo_arithmetic_integral_const.rs:23:5
   |
LL |     -1i64 % 2i64;
   |     ^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `1 % -2`
  --> $DIR/modulo_arithmetic_integral_const.rs:24:5
   |
LL |     1i64 % -2i64;
   |     ^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-1 % 2`
  --> $DIR/modulo_arithmetic_integral_const.rs:25:5
   |
LL |     -1i128 % 2i128;
   |     ^^^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `1 % -2`
  --> $DIR/modulo_arithmetic_integral_const.rs:26:5
   |
LL |     1i128 % -2i128;
   |     ^^^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `-1 % 2`
  --> $DIR/modulo_arithmetic_integral_const.rs:27:5
   |
LL |     -1isize % 2isize;
   |     ^^^^^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: you are using modulo operator on constants with different signs: `1 % -2`
  --> $DIR/modulo_arithmetic_integral_const.rs:28:5
   |
LL |     1isize % -2isize;
   |     ^^^^^^^^^^^^^^^^
   |
   = note: double check for expected result especially when interoperating with different languages
   = note: or consider using `rem_euclid` or similar function

error: aborting due to 17 previous errors

